webapp/params.go

```diff
@@ -95,6 +87,53 @@ func ParseBrowsersParam(r *http.Request) (browsers []string, err error) {
 	return browsers, nil
 }
 
+// GetBrowsersForRequest parses the 'browsers' param, returning the sorted list of browsers
+// to include, or a default list.
+func GetBrowsersForRequest(r *http.Request) (browsers []string, err error) {
+	if browsers, err = ParseBrowsersParam(r); err != nil {
+		return browsers, err
+	}
+	if browsers == nil || len(browsers) == 0 {
+		if browsers, err = GetBrowserNames(); err != nil {
+			return browsers, err
+		}
+	}
+	labels := ParseLabelsParam(r)
+	if labels != nil {
+		experimental := labels.Contains(experimentalLabel)
+		browserNames, err := GetBrowserNames()
+		if err != nil {
+			return nil, err
+		}
+
+		browserLabel := ""
+		for _, name := range browserNames {
+			if !labels.Contains(name) {
+				continue
+			}
+			// If we already encountered a browser name, nothing is two browsers (return empty set).
+			if browserLabel != "" {
+				browsers = nil
+				break
+			}
+			browserLabel = name
+			browsers = []string{name}
+			// For a browser label (e.g. "chrome"), we also include experimental, unless we explicitly only
```

Comment:
I think all of this logic works, but it does make it very easy/tempting to infer some labels from others, treat combinations of labels differently, and so on.

To avoid intentional or accidental things like that, would it make sense to instead compute a set of labels for each run/runner, and then filter those by labels in a label-name-agnostic way "here?" That does require the frontend to know about all the runs/runners that exist, which perhaps it doesn't, but wouldn't that make sense?

I ask that partially because I think there's a limitation in labels, namely that filtering to Edge Insider Preview + Chrome Stable won't work, and those may well be interesting to compare. Don't know what the fix is, but suspect it'll get weird if we try to add it to this structure.

Reply:
`TestRun` will eventually include a `Labels []string` concept, which the collection endpoint will leverage (things like browser, os, stable/experimental, and then manually flagged (bad/incomplete) etc.)

For the browser v browser use-case, there's a (repeated) `browser` param in the API, e.g. https://wpt.fyi/api/runs?browsers=chrome,firefox - I'll put together a PR to leverage that in (bubble it from) the wpt.fyi UI.

All that is to say, the hacky-looking logic here will die in a fire when _actual_ labels exist. But the URLs/usage will stay the same :)

Your comment, surmisable as roughly _should multiple labels be AND combined, or OR combined?_ is something worth considering. Or-wise searching is probably not something we need to worry about, though, since the labels can be thought of as 'groups' (stable browsers. chrome browsers. mobile browsers).

Reply:
Yes, that is roughly the shape of my question. Note that if there's a way to refer to a specific runner's latest run or a specific run, then this concern also goes poof, since what I'm really after is comparing Edge Insider Preview to Chrome Stable.

OK to let this be until real labels exist, and probably even then.